### PR TITLE
Add `AfterCommit` plugin

### PR DESCRIPTION
**Pull request by jcredding**
This adds a model, record and fake record plugin that allows adding
activerecord `after_commit` callbacks from outside the record. From
the model, a proc can be added as an after commit callback and it
will run once activerecord commits the transaction. This is useful
for adding logic to run after a transaction without having to
always have it as a callback.

This adds mixins for a model, record and fake record that expose
the after commit logic. The record installs after commit callbacks
that always attempt to run any callbacks that are added to the
record. The model demeters the records so callbacks can be easily
added using it. The fake record provides the same interface and
fake's the real record's behavior. This allows easily testing
models that use the after commit logic.

@kellyredding - Ready for review.

### CODE REVIEW

**Comment by kellyredding** [comment]
@jcredding I see you chose to put all things in one file.  Any reason why?

File: lib/mr/after_commit.rb [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+    end
+
+    module RecordProcsMethods
+
+      # these methods are used by the `Record` and `FakeRecord` mixins
+
+      def after_commit_procs(*keys)
+        if keys.empty?
+          mr_after_commit_procs_hash.values.flatten
```
Comment (kellyredding) [inline lib/mr/after_commit.rb]: @jcredding technically this is a departure in behavior from what you are extracting this from (it just returned the has).  Is this safe?  Is this intended?

**Comment by kellyredding** [comment]
@jcredding so this is pretty cool in general.  It's a little odd seeing so much crammed into a single file (I don't mind it necessarily, just seems odd).  I had a couple questions I would like to get your thoughts on but in general I'm cool with this. 👎

**Comment by jcredding** [comment]
@kellyredding - I ended up splitting this up into multiple files. I put it into a single file because I thought it make it simple to require and because of how small the logic seemed to be. So I chose to originally implement it that way. I didn't necessarily love it when I finished but was ok with it.

In coming back to it and looking at it I kept being less and less happy with it being in a single file and your correct it felt weird so I went ahead and split it up. I like this better and I still kept it simple to require (only have to `require 'mr/after_commit'`).

**Comment by kellyredding** [comment]
@jcredding yeah, I like this better too.  👎

**Pull request merged**